src/send/SendSelectRecipient.tsx

```diff
@@ -135,27 +152,167 @@ const getStartedStyles = StyleSheet.create({
+      defaultCountryCode ? defaultCountryCode : undefined
+    )
+
+    if (parsedPhoneNumber) {
+      debounceSearchQuery(parsedPhoneNumber.e164Number)
+    } else {
+      debounceSearchQuery(searchQuery)
+    }
+  }, [searchQuery, defaultCountryCode])
+  const { result: resolveAddressResult } = useAsync(resolveId, [debouncedSearchQuery])
+
+  const setSearchQueryWrapper = (query: string) => {
+    // Always unset the selected recipient and hide the send/invite button
+    // when the search query is changed in order to prevent edge cases
+    // where the button appears but is bound to a recipient that is
+    // not present on the page.
+    unsetSelectedRecipient()
+    setSendOrInviteButtonHidden(true)
+
+    if (!query) {
+      setShowSearchResults(false)
+    } else {
+      setShowSearchResults(true)
+    }
+    setSearchQuery(query)
+  }
+
+  const [showInviteModal, setShowInviteModal] = useState(false)
 
   const [showContacts, setShowContacts] = useState(false)
 
   const recentRecipients = useSelector((state) => state.send.recentRecipients)
   const contactsCache = useSelector(phoneRecipientCacheSelector)
+  const recipientInfo = useSelector(recipientInfoSelector)
+
   const contactRecipients = useMemo(
     () => sortRecipients(Object.values(contactsCache)),
     [contactsCache]
   )
+  const [contactsFiltered, setContactsFiltered] = useState(() =>
+    sortRecipients(Object.values(contactRecipients))
+  )
+  const [recentFiltered, setRecentFiltered] = useState(() => recentRecipients)
+
+  const recentRecipientsFilter = useMemo(
+    () => filterRecipientFactory(recentRecipients, false),
+    [recentRecipients]
+  )
+  const contactRecipientsFilter = useMemo(
+    () => filterRecipientFactory(Object.values(contactRecipients), true),
+    [contactRecipients]
+  )
+
+  const getUniqueSearchRecipient = (): Recipient | undefined => {
+    const parsedNumber = parsePhoneNumber(
+      searchQuery,
+      defaultCountryCode ? defaultCountryCode : undefined
+    )
+    if (parsedNumber) {
+      return {
+        displayNumber: parsedNumber.displayNumber,
+        e164PhoneNumber: parsedNumber.e164Number,
+        recipientType: RecipientType.PhoneNumber,
+        name: t('sendToMobileNumber'),
+      }
+    }
+    if (isValidAddress(searchQuery)) {
+      return getRecipientFromAddress(searchQuery.toLowerCase(), recipientInfo)
+    }
+  }
+
+  const mapResolutionToRecipient = (resolution: NameResolution): Recipient => {
+    const lowerCaseAddress = resolution.address.toLowerCase()
+    switch (resolution.kind) {
+      case ResolutionKind.Address:
+        return getRecipientFromAddress(lowerCaseAddress, recipientInfo)
+      case ResolutionKind.Nom:
+        return {
+          address: lowerCaseAddress,
+          name: t('nomSpaceRecipient', { name: resolution.name ?? searchQuery }),
+          recipientType: RecipientType.Nomspace,
+        }
+      default:
+        return getRecipientFromAddress(lowerCaseAddress, recipientInfo)
```

Comment:
```suggestion
        const otherResolutionKind: never = resolution.kind
        throw new Error(
          `should never happen-- unrecotnized resolution kind ${otherResolutionKind} should be caught at compile time`
        )
```
an idea. this will help us notice we need to add a new case if we ever add a resolution kind that isn't handled here